autoPyTorch/api/base_task.py

```diff
@@ -215,7 +248,23 @@ def build_pipeline(self, dataset_properties: Dict[str, Any]) -> BasePipeline:
         Returns:
 
         """
-        raise NotImplementedError
+
+        raise NotImplementedError("Function called on BaseTask, this can only be called by "
+                                  "specific task which is a child of the BaseTask")
+
+    @abstractmethod
+    def _create_dataset(self,
+                        X_train: Union[List, pd.DataFrame, np.ndarray],
+                        y_train: Union[List, pd.DataFrame, np.ndarray],
+                        X_test: Union[List, pd.DataFrame, np.ndarray],
+                        y_test: Union[List, pd.DataFrame, np.ndarray],
+                        resampling_strategy: Union[CrossValTypes, HoldoutValTypes] = HoldoutValTypes.holdout_validation,
+                        resampling_strategy_args: Optional[Dict[str, Any]] = None,
+                        dataset_name: Optional[str] = None,
+                        return_only: Optional[bool] = False
```

Comment:
Instead of having a return_only, maybe create it once? 

So with return only, every time you call this method you will have to create and save a new pipeline. Do you think it is better to:


1. The first thing this function does is check if load_datamanger from the backend is able to load a dataset, and if it is the case, it loads it rather than creating a new one. 
2. Else a dataset is created

What do you think?

Reply:
The reason I had it create and always save a dataset, is that in case we want to have a different dataset to fit a pipeline, i.e, once we have made a search, we can try and fit the best incumbent configuration to a larger dataset or a different related dataset. return_only allows us to use the same function to create a dataset, whenever we get new data, i.e, in search, fit and refit. As we are also returning the dataset after the fit_pipeline method, the same dataset can be used to predict and score using the pipeline. 

Do you still think it is better to load a datamanager from the backend and use it? If so, do you know of a way to achieve them both.